examples/advanced/keycloak-site-authentication/README.md

```diff
@@ -0,0 +1,102 @@
+# Example for Job-level authorization
+
+# Overview
+
+The purpose of this example is to demonstrate following features of NVFlare,
+
+1. Run NVFlare in secure mode
+2. Demonstrate the ability to integrate with KeyCloak third-party user authentication for managing the NVFlare system. 
+
+## System Requirements
+
+1. Install Python and set up a Virtual Environment,
+```
+python3 -m venv nvflare-env
+source nvflare-env/bin/activate
+```
+2. Install NVFlare
+```
+pip install nvflare
+```
+3. The example is part of the NVFlare source code. The source code can be obtained like this,
+```
+git clone https://github.com/NVIDIA/NVFlare.git
+```
+4. TLS requires domain names. Please add the following line in the `/etc/hosts` file,
+```
+127.0.0.1	server1
+```
+
+### Setup
+
+```
+nvflare poc prepare -i project.yml -c site_a site_b
+```
+
+All the startup kits will be generated in this folder,
+```
+/tmp/nvflare/poc/keycloak-site-authentication/prod_00
+```
+
+### Set up KeyCloak
+
+Download and install the KeyCloak following the instruction guide from https://www.keycloak.org/getting-started/getting-started-zip. Key steps for the installation:
+
+* Start the KeyCLoak by running "bin/kc.sh start-dev"
+* Set up the realm called "myrealm"
+* Setup the user "myuser@example.com"
+* Add the client "myclient"
+* The KeyCLoak will be running at http://localhost:8080/
+
+### Set up Client side authentication requirement
+
+```
+cp -r site/local/* /tmp/nvflare/poc/keycloak-site-authentication/prod_00/site_a/local/* 
+```
+This will set up site_a with additional user security authentication to the KeyCloak.
+
+### Set up Admin user authentication
+
+```
+cp -r admin/local/* /tmp/nvflare/poc/keycloak-site-authentication/prod_00/myuser@example.co/local/* 
+```
+This will set up admin user myuser@example.com to acquire KeyCloak access_token when starting the admin tool.
```

Comment:
We should explain the content in the config file.

Reply:
added.